clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -6784,6 +6784,10 @@ def warn_arc_lifetime_result_type : Warning<
   "ARC %select{unused|__unsafe_unretained|__strong|__weak|__autoreleasing}0 "
   "lifetime qualifier on return type is ignored">,
   InGroup<IgnoredQualifiers>;
+def warn_next_larger_fp_type_same_size_than_fp : Warning<
+  "higher precision floating-point type requested by user size has the same size"
```

Comment:
This warning doesn't make sense to me.  Ignoring the weird grammar, the user didn't choose the higher precision type. The compiler chooses how it does the computation.

If the issue here that the target doesn't support a higher precision floating-point type, we should just say that.

Reply:
Could we diagnose from the driver or when we see a pragma requesting the higher-precision type? That would also reduce the chattiness of the diagnostic, I expect.

Reply:
This warning should trigger only for a division. When an `-fcomplex-arithmetic=promoted` is met, we don't have that knowledge.

Reply:
> This warning doesn't make sense to me. Ignoring the weird grammar, the user didn't choose the higher precision type. The compiler chooses how it does the computation.
> 
> If the issue here that the target doesn't support a higher precision floating-point type, we should just say that.

Yes, I can change the diagnostic comment to say that.

Reply:
> This warning should trigger only for a division. When an `-fcomplex-arithmetic=promoted` is met, we don't have that knowledge.

I realize that for division it causes a difference in program behavior, so division is a very important case to let users know about. However, the excess precision still isn't used for the other operations, right? So it's sort of a "you specified this flag but that flag doesn't ever do anything" situation?